clang/lib/CIR/CodeGen/CIRGenModule.cpp

```diff
@@ -347,7 +367,18 @@ void CIRGenModule::buildGlobal(GlobalDecl GD) {
   assert(!Global->hasAttr<IFuncAttr>() && "NYI");
   assert(!Global->hasAttr<CPUDispatchAttr>() && "NYI");
   assert(!langOpts.CUDA && "NYI");
-  assert(!langOpts.OpenMP && "NYI");
+
+  if (langOpts.OpenMP) {
```

Comment:
How about checking `OpenMPRuntime && OpenMPRuntime->emitTargetGlobal(GD)`? We should `assert(!UnimplementedFeature::openMPRuntime());` here, or something along these lines.

Reply:
Thanks for improving this. How about my comment on `OpenMPRuntime->emitTargetGlobal(GD)`? This should be similar in checks with what traditional LLVM is doing.

Reply:
My bad, I added the comment in the commit message. I left the `TODO[OpenMP]:` comments in `CIRGenExpr.cpp` as it is necessary to remove those assertions to test any patch using variables and that specific issue needs to be addressed quickly in a different patch on captured variables. Right now what it's doing it's treating all variables as shared. So either I change the test, or I left the TODOS.

Reply:
Yeas, totally makes sense, I actually wasn't sure why you had used `!` because it wouldn't assert. I'll change it right away.